senders/prometheus.go

```diff
@@ -173,12 +176,12 @@ func (s *PrometheusSender) buildLabelsFromMetric(metric metrics.Metric) (labelNa
 	return labelNames
 }
 
-func (s PrometheusSender) AppCreated(guid string) error {
+func (s PrometheusSender) AppInstanceCreated(guidInstance string) error {
 	return nil
 }
 
-func (s PrometheusSender) AppDeleted(guid string) error {
-	appMetrics := s.appMetrics[guid]
+func (s PrometheusSender) AppInstanceDeleted(guidInstance string) error {
+	appMetrics := s.appMetrics[guidInstance]
```

Comment:
But I have some doubts. Specifically, I see that the logic to notify about instances scale up and down is there, and that the "sender" gets those messages so it deletes the `appMetrics` vector based on the key `<guid>:<instance>`

But if I am reading right, that map, it is only created using only  the app guid, without the instance: `guid`:


- https://github.com/alphagov/paas-metric-exporter/blob/415ed0e25ccb348a483ef846cd5ee77e27ac2f15/senders/prometheus.go#L36-L37
- https://github.com/alphagov/paas-metric-exporter/blob/415ed0e25ccb348a483ef846cd5ee77e27ac2f15/senders/prometheus.go#L67-L68
- https://github.com/alphagov/paas-metric-exporter/blob/415ed0e25ccb348a483ef846cd5ee77e27ac2f15/senders/prometheus.go#L98-L99

now it is : `appMetrics := s.getOrCreateAppMetrics(metric.GUID)` but I guess it should be something like `appMetrics := s.getOrCreateAppMetrics(metric.GUID+":"metric.Instance)` (obviously something more elegant than that if possible :)).

Reply:
Said this, I think we should add a test around this feature. Maybe something like:

``` 
Given we get metrics for an app with 3 instances (metrics from each container)
Then prometheus fetcher should keep the metrics for the 3 instances
When the app is scaled down to 1 instance
Then prometheus fetcher should not keep the metrics for the extra instances.
```

Reply:
We've adjusted the stuff in the `senders/prometheus.go` file to take both the guid and instance and have added a test to `prometheus_test.go` to check that the metrics themselves have been removed. We think this resolves these comments.